src/config.rs

```diff
@@ -1094,4 +1148,61 @@ pub mod tests {
             assert!(metadata.vec.contains(&(key, value)));
         }
     }
+
+    #[test]
+    fn test_parse_worker_threads() {
```

Comment:
Will we flake if `test_parse_worker_threads` and `test_get_cpu_count` run in parallel since they are both setting/removing the same env vars and making assertions based on the contents?

Reply:
yes it does. will need to fix.

We should probably have some form of 'env var provider' abstraction

Reply:
Short term we could probably combine them into a single test to work around it. Little wonky but probably ok